packages/client/src/publish/ProxyPublisher.ts

```diff
@@ -0,0 +1,39 @@
+import { StreamDefinition } from '../types'
+import { StreamIDBuilder } from '../StreamIDBuilder'
+import { inject, Lifecycle, scoped } from 'tsyringe'
+import BrubeckNode from '../BrubeckNode'
+import { Context } from '../utils/Context'
+
+@scoped(Lifecycle.ContainerScoped)
+export default class ProxyPublisher {
+    constructor(
+        context: Context,
+        private node: BrubeckNode,
+        @inject(StreamIDBuilder) private streamIdBuilder: StreamIDBuilder,
+    ) {
+    }
+
+    async setPublishProxy(streamDefinition: StreamDefinition, nodeId: string): Promise<void> {
+        const streamPartId = await this.streamIdBuilder.toStreamPartID(streamDefinition)
+        await this.node.openPublishProxyConnectionOnStreamPart(streamPartId, nodeId)
+    }
+
+    async removePublishProxy(streamDefinition: StreamDefinition, nodeId: string): Promise<void> {
+        const streamPartId = await this.streamIdBuilder.toStreamPartID(streamDefinition)
+        await this.node.closePublishProxyConnectionOnStreamPart(streamPartId, nodeId)
+    }
+
+    async setPublishProxies(streamDefinition: StreamDefinition, nodeIds: string[]): Promise<void> {
```

Comment:
To keep the public API simple, we could remove the singular form of methods (`setPublishProxy` and `removePublishProxy`) and keep only the plural forms. It is trivial to provide a single-item list as a `nodeIds` argument. We did similar change in client with permission-related methods.